sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs

```diff
@@ -74,21 +74,22 @@ private void ReportOnViolation(SyntaxNodeAnalysisContext context)
                 .Cast<LiteralExpressionSyntax>();
 
             // Collect duplications
-            var stringWithLiterals = new Dictionary<string, List<LiteralExpressionSyntax>>();
+            var stringWithLiterals = new Dictionary<string, List<Location>>();
             foreach (var literal in stringLiterals)
             {
-                var stringValue = literal.Token.ValueText;
+                // Remove leading and trailing double quotes
+                var stringValue = literal.Token.Text.Substring(1, literal.Token.Text.Length - 2);
```

Comment:
Actual change.

Reply:
Does it work with strings like `@"asdasd"` or `$@"adasdasd"`? (not sure if those are different syntax kinds)

Reply:
`$@""` and `$""` are not `StringLiteralExpression` and so not handled by this rule.